### ISSUE TITLE
[eslint-plugin-lit-a11y]: implement media-has-caption

### ISSUE THREAD

**Issue by bennypowers** [open-wc/open-wc#1926]
See: https://raw.githubusercontent.com/jsx-eslint/eslint-plugin-jsx-a11y/master/docs/rules/media-has-caption.md

# media-has-caption

Providing captions for media is essential for deaf users to follow along. Captions should be a transcription or translation of the dialogue, sound effects, relevant musical cues, and other relevant audio information. Not only is this important for accessibility, but can also be useful for all users in the case that the media is unavailable (similar to `alt` text on an image when an image is unable to load).

The captions should contain all important and relevant information to understand the corresponding media. This may mean that the captions are not a 1:1 mapping of the dialogue in the media content. However, captions are *not* necessary for video components with the `muted` attribute.

## Rule details

This rule takes one optional object argument of type object:

```json
{
    "rules": {
        "lit-a11y/media-has-caption": ["error", {
            "audio": [
              "custom-audio",
              {
                "tagName": "slotted-audio",
                "slot": ["track", "caption"]
              }
              {
                "tagName": "custom-multimedia",
                "slot": ["audio-caption"]
              }
            ],
            "video": [
              {
                "tagName": "custom-multimedia",
                "slot": ["video-caption"]
              }
            ],
          }],
    }
}
```

For the `audio` and `video` options, strings determine which custom elements should be considered to implement the corresponding media element internally.
If a string is specified, the custom element is assumed to implement an anonymous slot to it's internal media element.
If an object is specified, `track` elements must be slotted into 

### Succeed
```js
html`
  <audio><track kind="captions"></track></audio>
  <video><track kind="captions"></track></video>
  <video muted></video>
`;
```

### Fail
```js
html`
  <audio></audio>
  <video></audio>
`
```

## Accessibility guidelines
- [WCAG 1.2.2](https://www.w3.org/WAI/WCAG21/Understanding/captions-prerecorded.html)
- [WCAG 1.2.3](https://www.w3.org/WAI/WCAG21/Understanding/audio-description-or-media-alternative-prerecorded.html)

### Resources
- [axe-core, audio-caption](https://dequeuniversity.com/rules/axe/2.1/audio-caption)
- [axe-core, video-caption](https://dequeuniversity.com/rules/axe/2.1/video-caption)
g
